internal/pkg/selfupdate/update.go

```diff
@@ -15,9 +15,10 @@ import (
 	"github.com/cli/safeexec"
 )
 
-func UpgradeInPlace(ctx context.Context, baseSiteURL string, channelName *string, stdout, stderr io.Writer, stdin io.Reader, detach bool) error {
+func UpgradeInPlace(ctx context.Context, curSemanticVersion string, baseSiteURL string, channelName *string, stdout, stderr io.Writer, stdin io.Reader, detach bool) error {
 
-	cur, renamed, err := renameCurrentBinaryWithPostfix(".old")
+	curSemanticVersion = strings.ReplaceAll(curSemanticVersion, ".", "-")
```

Comment:
instead of doing this here, make sure that the string created from the semver object is valid in the first place